### PR TITLE
Recipe maintainership bump

### PR DESCRIPTION
**Pull request by h-vetinari**
Checklist
* [x] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

I suggested to the cvxpy-maintainers to [consider](https://github.com/cvxgrp/cvxpy/issues/937#issuecomment-583855932) moving their existing conda-packaging to conda-forge, and they after some deliberation, @SteveDiamond, @akshayka & @rileyjmurray [confirmed](https://github.com/cvxgrp/cvxpy/issues/937#issuecomment-606327541) that they would like to do that. This is the first commit (@conda-forge/cvxpy was involved in the discussion there already).

Also, after a few PRs on this feedstock, @jakirkham [asked](https://github.com/conda-forge/cvxpy-feedstock/pull/30#issuecomment-606303529) me if I want to join as a maintainer as well - this is the second commit, but I'm happy to take it out if people want to have this separate.

@conda-forge-admin, please rerender

### CODE REVIEW

**Comment by scopatz** [comment]
@SteveDiamond, @akshayka & @rileyjmurray - can you all please confirm here that you would like to be added as maintainers?

**Comment by SteveDiamond** [comment]
@scopatz I confirm that I want to be a maintainer.

**Comment by h-vetinari** [comment]
Failures are just timeouts of `https://repo.anaconda.com`. The exact same build config ran through on #31, so I'm not worried about that.

**Comment by akshayka** [comment]
Sure, I confirm as well.

**Comment by rileyjmurray** [comment]
@scopatz yes, please add me as a maintainer.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ source:
     - patches/0001-remove-pyproject.toml-requires-in-setup.py.patch
 
 build:
-  number: 0
+  number: 1
```
Comment (scopatz) [inline recipe/meta.yaml:15]: This does not need a version number bump.  Please reset to zero

Comment (h-vetinari) [reply recipe/meta.yaml:15]: I know that the build doesn't change, but I wasn't sure what happens e.g. if build jobs time out after merging to master. Removing

**Comment by h-vetinari** [comment]
@conda-forge-admin, please rerender

**Comment by h-vetinari** [comment]
Ping @scopatz @jakirkham 
Should be good to go, all feedback incorporated and the build is green.

**Comment by scopatz** [comment]
Thanks all

**Pull request merged**